command/pr.go

```diff
@@ -206,8 +207,10 @@ func prList(cmd *cobra.Command, args []string) error {
 	})
 
 	title := listHeader(ghrepo.FullName(baseRepo), "pull request", len(listResult.PullRequests), listResult.TotalCount, hasFilters)
-	// TODO: avoid printing header if piped to a script
-	fmt.Fprintf(colorableErr(cmd), "\n%s\n\n", title)
+
+	if utils.IsTerminal(os.Stdout) {
```

Comment:
Instead of using global `os.Stdout`, we reference the output stream inside commands as `cmd.OutOrStdout()`, so you should be checking that object instead. This gives us the ability to override the output stream for tests. Note that the output stream is a `Writer` and not necessarily a `File`, so you would have to do an additional check.

Reply:
Fares